test/core/test/jest-mock.test.ts

```diff
@@ -88,54 +87,6 @@ describe('jest mock compat layer', () => {
     expect(spy.mock.contexts).toEqual([instance])
   })
 
-  it('throws an error when constructing a class with an arrow function', () => {
```

Comment:
@AriPerkkio  I decided to revert this change because it interferes with the user code. We can't really predict why the error happened - it might be an error in the source code somewhere. And if so, they won't be able to properly catch it.

But we can maybe print a warning in this case?